packages/editor/src/store/actions/index.js

```diff
@@ -0,0 +1,2 @@
+export * from './post-actions';
```

Comment:
Is there a compelling reason to segregate the two?

Reply:
Do you mean among the post actions?  or do you mean specifically having "post" type actions and eventually "block" type actions?

If the former, I find it easier to know where to go for general actions as opposed to action generators.  It also aids with splitting up tests to cover the specific things.

Reply:
> I find it easier to know where to go for general actions as opposed to action generators.

It doesn't seem to me that it's something someone should need to distinguish. I think a current reality is that we have spaghetti generators, but they're all just "actions (dispatchers)", ideally returning or yielding simple action objects.

Reply:
I'm probably ignorant of potential reasons why we wouldn't want to split up things in this case.  Are there technical/performance/build reasons why its good to avoid splitting up files like this?

Reply:
I usually try to logically split files up into smaller components when I can because:

a. it helps with finding things quicker when I'm just looking for a "class" of logic
b. it makes for smaller number of lines in a file, so less scrolling.
c. My IDE built-in linting etc doesn't get bogged down as much due to larger files.

But as I already mentioned, I'm aware I may be ignorant of good reasons for not splitting things up as I did here.

Reply:
To me, it's not a technical point, it's of discoverability and maintainability generally. What we're doing here is creating a subclass (of actions) where otherwise we've aligned to a common pattern of identifying a few top-level behaviors associated with a store: mutations (`actions.js`), state (`reducer.js`), derivations / getters (`selectors.js`), and side-effects (`effects.js`).

I'm not saying this can't be expanded, but we should do so carefully, deliberately, and consistently. Small files are not inherently more maintainable than a single large file, if it's not obvious or predictable why the files are split the way they are. I find the `effects/` folder pattern to be a good example of this, where I think it was better off in single file form, because the current split is not obvious, and in the majority of cases I arrive at the wrong place when trying to track down effects-related behavior.

My prior point was that the division between generator-actions and non-generator-actions is a bit weak, and arguably intentionally so (my point about "you shouldn't need to distinguish"), so creating subclasses within actions for these does not seem an obvious thing to do.

Reply:
Kinda funny that we're both hinging our argument on discoverability and maintainability :).  I'm not a project maintainer so I don't want to break any established conventions for the project.  I'm happy to just combine them into one file (and the same with tests) if desired.  So just for clarification, is this a _directive_ I need to follow in order to get this pull mergeable or is this a preference you are arguing for but is not a directive yet?

To be clear, I did loosely swing my intentions by @youknowriad a bit in dm with him before embarking, although he suggested this structure (which mirrors the current effects):

> Personally, I'd start with adding those to `actions.js` but maybe split it by domains
>
> /actions/index.js
> /actions/post.js
> /actions/reusable-blocks.js

So I _did_ expand on that a bit in my implementation.  So in your clarification, can you also give direction for whether I should just put everything into the existing `actions.js` file (and subsequently add all the tests to the existing `tests/actions.js` as well?

Reply:
Nope, no directives. Just surfacing my own opinions stemming from the initial concern / probe for rationale of the split. It seems there's at least a second opinion in your favor from @youknowriad , but I'd go on record to disagree with the "domain"-based split (it wasn't but an hour after my previous complaint about the effects structuring that I wrongly navigated to `effects/posts.js` as my first destination in pursuit of [this snippet of code](https://github.com/WordPress/gutenberg/blob/a99c0b5487ff3639c05a7ce1dbf7823b821382f8/packages/editor/src/store/effects.js#L206-L216)).

To your question, yes, I'd see it all just being in a single `actions.js` file.

There's an interesting overlap between this discussion and some older ones like in #3030, where I think ultimately it was the introduction of `@wordpress/data` and its store-namespacing that served as the alternative to this problem of files becoming large and unwieldy, where the separation was by _stores_ (modules), not by separate files within a store.

Reply:
`actions.js` is better if we had one file but if the file grows indefinitely, it becomes hard to navigate the file. maybe there's a middleground to find at some point.

I also said to @nerrad that in general I'm not very opinionated on these matters :)

Reply:
> but if the file grows indefinitely, it becomes hard to navigate the file.

Ya this is one criteria I use when deciding whether to split up a file.  For example _just_ with the post actions here, combined, they would be 487 lines.  That's _before_ adding in all the block and editor related actions.

> as my first destination in pursuit of [this snippet of code](https://github.com/WordPress/gutenberg/blob/a99c0b5487ff3639c05a7ce1dbf7823b821382f8/packages/editor/src/store/effects.js#L206-L216)).

@aduth my _plan_ was to have `editor-actions` and `block-actions` files that would accompany the `post-actions`.  That snippet is found within the `SETUP_EDITOR` editor block. 

Would a middle ground here be to not split between action generators and action functions and just have `post-actions`, `editor-actions`, and `block-actions`?  That might also be useful for aiding future refactors where some (or all) of the block specific actions are moved elsewhere.

Reply:
Keep in mind, I'm not only considering the line count for a combined actions file, but also the line count that would be exponentially larger for the corresponding tests file.  Currently combining `actions/test/post.js` and `actions/test/post-generators.js` into one file results in 759 lines.  As soon as a file gets over roughly 500 lines I find it starts to get unwieldy.  I imagine the line count for a combined actions file of editor, post, and block actions will start to go over 2000 easily.

Reply:
> There's an interesting overlap between this discussion and some older ones like in #3030, where I think ultimately it was the introduction of `@wordpress/data` and its store-namespacing that served as the alternative to this problem of files becoming large and unwieldy, where the separation was by _stores_ (modules), not by separate files within a store.

To explore this further then, are you suggesting that we might want to consider applying some more granular namespacing here and split these up into more stores?  Of course, that could introduce some complications with deprecating the old actions etc but might be preferable over the file splitting.  So we could end up with something like `core/editor/posts`, `core/editor` (for general editor actions), and `core/editor/blocks` for block actions?